Sources/AppcuesKit/Presentation/Traits/Appcues/ExperienceWrapperViewController+DragDirection.swift

```diff
@@ -0,0 +1,91 @@
+//
+//  ExperienceWrapperViewController+DragDirection.swift
+//  AppcuesKit
+//
+//  Created by Matt on 2024-09-27.
+//  Copyright © 2024 Appcues. All rights reserved.
+//
+
+import SwiftUI
+
+@available(iOS 13.0, *)
+extension ExperienceWrapperViewController {
+    struct DragDirection {
+        enum Edge {
+            case left, right, up, down
+        }
+
+        let xDirection: Edge
+        let yDirection: Edge
+
+        private let initialPoint: CGPoint
+        private let newPoint: CGPoint
+
+        init(initial: CGPoint, new: CGPoint) {
+            self.initialPoint = initial
+            self.newPoint = new
+            self.xDirection = new.x > initial.x ? .right : .left
+            self.yDirection = new.y > initial.y ? .down : .up
+        }
+
+        /// Determine which edges are eligible for drag-to-dismiss.
+        static func allowedEdges(horizontalAlignment: HorizontalAlignment, verticalAlignment: VerticalAlignment) -> Set<Edge> {
```

Comment:
is this a Set for future extensibility to allow dragging away to multiple edges?

Reply:
Hah kinda... It's a set because I started off allowing dragging to multiple edges and then tweaked it to be restricted to a single edge since that felt better. For slideouts in the corners, ie bottom left, it's not the worst to allow it to be swiped to the left or the bottom, but my personal preference is only to the left